File: ios-native/Dimo/Features/Email/EmailScreen.swift
import SwiftUI

struct EmailScreen: View {
  @Bindable var store: EmailFeatureStore
  var onOpenSettings: () -> Void

  var body: some View {
    VStack(spacing: 0) {
      header

      ScrollView {
        LazyVStack(alignment: .leading, spacing: 16) {
          if store.accounts.isEmpty {
            connectAccountCard
          } else {
            accountAndAnalyzerStrip
            filters

            if store.selectedFilter.displaysMessages {
              if store.filteredEmails.isEmpty {
                emptyState
              } else {
                ForEach(store.filteredEmails) { email in
                  EmailMessageStatusCard(
                    email: email,
                    onOpen: { store.presentEmail(id: email.id) },
                    onRestore: email.analysisState == .dismissed
                      ? { store.restoreSuggestion(email.id) }
                      : nil,
                    onRetry: email.analysisState == .failed
                      ? { store.retryAnalysis(messageID: email.id) }
                      : nil,
                    onRetryWithAlternate: email.analysisState == .failed
                      ? { store.retryWithAlternateProvider(messageID: email.id) }
                      : nil
                  )
                }
              }
            } else if store.filteredSuggestions.isEmpty {
              emptyState
            } else {
              ForEach(store.filteredSuggestions) { suggestion in
                EmailSuggestionCard(
                  suggestion: suggestion,
                  onOpen: { store.presentEmail(id: suggestion.id) },
                  onReview: { store.review(suggestion) },
                  onDismiss: { store.dismissSuggestion(suggestion.id) },
                  onRestore: suggestion.status == .dismissed
                    ? { store.restoreSuggestion(suggestion.id) }
                    : nil
                )
              }
            }

            privacyNote
          }
        }
        // Message and suggestion rows share the same underlying email key as
        // their ForEach identity. Re-key the lazy container per filter so it
        // never reuses cached cells of the other card type across tab switches.
        .id(store.selectedFilter)
        .padding(.horizontal, 22)
        .padding(.top, 10)
        .padding(.bottom, 34)
      }
    }
    .background(Theme.canvas.ignoresSafeArea())
    .sheet(item: $store.emailDetail) { detail in
      EmailDetailSheet(detail: detail, onClose: store.dismissEmailDetail)
    }
    .sheet(item: $store.refundReview) { initialReview in
      RefundReviewSheet(
        review: Binding(
          get: { store.refundReview ?? initialReview },
          set: { store.refundReview = $0 }
        ),
        activeCurrency: store.activeCurrency,
        onCancel: { store.refundReview = nil },
        onMarkReviewed: {
          store.dismissSuggestion($0.suggestionID)
          store.refundReview = nil
        },
        onConfirm: { store.applyFullRefund($0) }
      )
    }
    .alert(
      "Email action failed",
      isPresented: Binding(
        get: { store.lastActionError != nil },
        set: { if !$0 { store.clearError() } }
      )
    ) {
      Button("OK") { store.clearError() }
    } message: {
      Text(store.lastActionError ?? "Please try again.")
    }
  }

  private var header: some View {
    HStack(spacing: 12) {
      VStack(alignment: .leading, spacing: 2) {
        Text("Email")
          .font(DimoFont.display(24, weight: .semibold))
          .foregroundStyle(Theme.ink)
        Text("Purchase and refund suggestions")
          .font(DimoFont.body(12))
          .foregroundStyle(Theme.muted)
      }

      Spacer()

      if store.hasFailedAnalyses {
        Button(action: store.reanalyzeAllEmails) {
          Image(systemName: "arrow.clockwise")
            .font(.system(size: 18, weight: .semibold))
            .foregroundStyle(Theme.ink)
            .frame(width: 42, height: 42)
            .background(Theme.surface)
            .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 13, style: .continuous).stroke(Theme.line))
        }
        .buttonStyle(.plain)
        .disabled(store.isReanalyzing)
        .accessibilityLabel(
          store.isReanalyzing ? "Reanalysing failed emails" : "Reanalyse failed emails"
        )
      }

      Button(action: onOpenSettings) {
        Image(systemName: "gearshape")
          .font(.system(size: 18, weight: .semibold))
          .foregroundStyle(Theme.ink)
          .frame(width: 42, height: 42)
          .background(Theme.surface)
          .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
          .overlay(RoundedRectangle(cornerRadius: 13, style: .continuous).stroke(Theme.line))
      }
      .buttonStyle(.plain)
      .accessibilityLabel("Email accounts and analyzer settings")
    }
    .frame(minHeight: 64)
    .padding(.horizontal, 22)
    .padding(.top, 8)
  }

  private var connectAccountCard: some View {
    VStack(alignment: .leading, spacing: 14) {
      Image(systemName: "envelope.badge")
        .font(.system(size: 24, weight: .semibold))
        .foregroundStyle(Theme.green)
        .frame(width: 48, height: 48)
        .background(Theme.greenSoft)
        .clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous))

      Text("Find expenses in Gmail")
        .font(DimoFont.display(20, weight: .semibold))
        .foregroundStyle(Theme.ink)

      Text("Connect one or more Gmail accounts. Dimo reads the latest \(store.syncWindow.title) directly on this iPhone. You choose Local Gemma or OpenRouter for analysis; analyzed suggestions sync through Dimo for restore.")
        .font(DimoFont.body(14))
        .foregroundStyle(Theme.body)
        .fixedSize(horizontal: false, vertical: true)

      ActionButton(title: "Connect Gmail", variant: .accent) {
        store.connectAccount()
      }

      Text("Read-only Gmail access. Gmail credentials stay on this iPhone. You approve every transaction change.")
        .font(DimoFont.body(11))
        .foregroundStyle(Theme.muted)
        .frame(maxWidth: .infinity, alignment: .center)
    }
    .padding(20)
    .background(Theme.surface)
    .clipShape(RoundedRectangle(cornerRadius: 20, style: .continuous))
    .overlay(RoundedRectangle(cornerRadius: 20, style: .continuous).stroke(Theme.line))
  }

  private var accountAndAnalyzerStrip: some View {
    HStack(spacing: 8) {
      ScrollView(.horizontal, showsIndicators: false) {
        HStack(spacing: 8) {
          ForEach(store.accounts) { account in
            Button {
              store.refreshAccount(account.id)
            } label: {
              HStack(spacing: 8) {
                accountRefreshIcon(account.syncState)
                VStack(alignment: .leading, spacing: 1) {
                  Text(account.emailAddress)
                    .font(DimoFont.body(11, weight: .medium))
                    .foregroundStyle(Theme.ink)
                    .lineLimit(1)
                  Text(account.statusDetail ?? account.syncState.title)
                    .font(DimoFont.body(10))
                    .foregroundStyle(account.syncState == .failed ? Theme.danger : Theme.muted)
                    .lineLimit(1)
                }
              }
              .padding(.horizontal, 10)
              .frame(height: 52)
              .background(Theme.surface)
              .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
              .overlay(RoundedRectangle(cornerRadius: 13, style: .continuous).stroke(Theme.line))
            }
            .buttonStyle(.plain)
            .disabled(account.syncState == .syncing)
            .accessibilityHint("Refresh this Gmail account")
          }
        }
      }
      .frame(maxWidth: .infinity)

      Button(action: onOpenSettings) {
        HStack(spacing: 8) {
          Image(systemName: analyzerIcon)
            .font(.system(size: 13, weight: .semibold))
            .foregroundStyle(store.selectedProvider != nil ? Theme.green : Theme.muted)
            .frame(width: 28, height: 28)
            .background(store.selectedProvider != nil ? Theme.greenSoft : Theme.canvasDeep)
            .clipShape(RoundedRectangle(cornerRadius: 9, style: .continuous))

          VStack(alignment: .leading, spacing: 1) {
            Text(store.activeAnalyzerTitle)
              .font(DimoFont.body(11, weight: .semibold))
              .foregroundStyle(Theme.ink)
              .lineLimit(1)
            Text(compactAnalyzerDetail)
              .font(DimoFont.body(10))
              .foregroundStyle(Theme.muted)
              .lineLimit(1)
          }

          if let progress = store.modelState.progress {
            Text(progress.formatted(.percent.precision(.fractionLength(0))))
              .font(DimoFont.body(10, weight: .semibold))
              .foregroundStyle(Theme.green)
          }

          Image(systemName: "chevron.right")
            .font(.system(size: 10, weight: .semibold))
            .foregroundStyle(Theme.faint)
        }
        .padding(.horizontal, 10)
        .frame(height: 52)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(Theme.surface)
        .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 13, style: .continuous).stroke(Theme.line))
      }
      .buttonStyle(.plain)
      .frame(maxWidth: .infinity)
      .accessibilityHint("Choose and configure the email analyzer")
    }
  }

  private func accountRefreshIcon(_ state: EmailUIAccountSyncState) -> some View {
    Group {
      if state == .syncing {
        ProgressView().controlSize(.mini).tint(Theme.green)
      } else {
        Image(systemName: state == .failed ? "exclamationmark.arrow.circlepath" : "arrow.clockwise")
          .font(.system(size: 13, weight: .semibold))
          .foregroundStyle(state == .failed ? Theme.danger : Theme.green)
      }
    }
    .frame(width: 28, height: 28)
    .background(state == .failed ? Theme.danger.opacity(0.12) : Theme.greenSoft)
    .clipShape(RoundedRectangle(cornerRadius: 9, style: .continuous))
    .accessibilityHidden(true)
  }

  private var compactAnalyzerDetail: String {
    if store.selectedProvider != .gemma {
      return store.analysisStatusDetail
    }
    switch store.modelState {
    case .notInstalled:
      return "Tap to download"
    case .checkingStorage:
      return "Checking storage…"
    case .downloading:
      return "Downloading…"
    case .paused:
      return "Download paused"
    case .verifying:
      return "Verifying…"
    case .installed(let version):
      if let detail = store.gemmaStatusDetail {
        return detail
      }
      return store.isGemmaAnalyzerAvailable ? "v\(version)" : "Analysis unavailable"
    case .failed(let message), .unavailable(let message):
      return message
    }
  }

  private var analyzerIcon: String {
    switch store.selectedProvider {
    case .gemma: return "cpu"
    case .openRouter: return "cloud"
    case nil: return "text.magnifyingglass"
    }
  }

  private var filters: some View {
    ScrollView(.horizontal, showsIndicators: false) {
      HStack(spacing: 8) {
        ForEach(EmailSuggestionFilter.allCases) { filter in
          Button {
            withAnimation(.easeOut(duration: 0.18)) {
              store.selectedFilter = filter
            }
          } label: {
            HStack(spacing: 6) {
              Text(filter.title)
              let count = suggestionCount(for: filter)
              if count > 0 {
                Text("\(count)")
                  .font(.system(size: 10, weight: .bold))
                  .padding(.horizontal, 5)
                  .padding(.vertical, 2)
                  .background(store.selectedFilter == filter ? Theme.canvas.opacity(0.2) : Theme.canvasDeep)
                  .clipShape(Capsule())
              }
            }
            .font(DimoFont.body(12, weight: .medium))
            .foregroundStyle(store.selectedFilter == filter ? Theme.canvas : Theme.ink)
            .padding(.horizontal, 14)
            .frame(height: 38)
            .background(store.selectedFilter == filter ? Theme.ink : Theme.surface)
            .clipShape(Capsule())
            .overlay(Capsule().stroke(Theme.line, lineWidth: store.selectedFilter == filter ? 0 : 1))
          }
          .buttonStyle(.plain)
        }
      }
    }
    .accessibilityElement(children: .contain)
  }

  private var emptyState: some View {
    VStack(spacing: 10) {
      Image(systemName: emptyStateIcon)
        .font(.system(size: 28, weight: .medium))
        .foregroundStyle(Theme.faint)
      Text(emptyStateTitle)
        .font(DimoFont.body(15, weight: .semibold))
        .foregroundStyle(Theme.ink)
      Text(emptyStateDetail)
        .font(DimoFont.body(13))
        .foregroundStyle(Theme.muted)
        .multilineTextAlignment(.center)
        .fixedSize(horizontal: false, vertical: true)
    }
    .frame(maxWidth: .infinity)
    .padding(.horizontal, 24)
    .padding(.vertical, 44)
  }

  private var emptyStateIcon: String {
    switch store.selectedFilter {
    case .all: return "tray.full"
    case .purchases: return "cart"
    case .refunds: return "arrow.uturn.backward.circle"
    case .awaitingAnalysis: return "hourglass"
    case .reviewed: return "checkmark.circle"
    }
  }

  private var emptyStateTitle: String {
    switch store.selectedFilter {
    case .all: return "No scanned emails yet"
    case .purchases: return "No purchase suggestions"
    case .refunds: return "No refund suggestions"
    case .awaitingAnalysis: return "No emails awaiting analysis"
    case .reviewed: return "Nothing reviewed yet"
    }
  }

  private var emptyStateDetail: String {
    switch store.selectedFilter {
    case .all:
      return "Scanned messages and their local analysis status will appear here."
    case .purchases, .refunds:
      return "Email scanning is best effort and is not real time. Tap an account above to refresh."
    case .awaitingAnalysis:
      return "Fetched emails waiting for analysis will appear here."
    case .reviewed:
      return "Suggestions you add, dismiss, or apply will appear here."
    }
  }

  private var privacyNote: some View {
    Label(privacyNoteText, systemImage: "lock.shield.fill")
    .font(DimoFont.body(11))
    .foregroundStyle(Theme.muted)
    .fixedSize(horizontal: false, vertical: true)
    .padding(.top, 4)
  }

  private var privacyNoteText: String {
    switch store.selectedProvider {
    case .gemma:
      return "Local Gemma analyzes email on this iPhone. Gmail credentials stay on-device. Analyzed suggestions and their email text sync through Dimo so they restore across your devices. You still approve every transaction change."
    case .openRouter:
      return "Analysis goes from this iPhone to OpenRouter and the selected provider. Analyzed suggestions and their email text then sync through Dimo so they restore across your devices. You still approve every transaction change."
    case nil:
      return "Gmail credentials and pending email content stay on this iPhone until you configure an analyzer. Analyzed suggestions later sync through Dimo for restore."
    }
  }

  private func suggestionCount(for filter: EmailSuggestionFilter) -> Int {
    switch filter {
    case .all:
      return store.allEmails.count
    case .purchases:
      return store.suggestions.count { suggestion in
        if suggestion.status == .pendingPurchase,
           suggestion.kind == .purchase || suggestion.kind == .debit {
          return true
        }
        return false
      }
    case .refunds:
      return store.suggestions.count {
        $0.status == .pendingRefund && $0.kind == .refund
      }
    case .awaitingAnalysis:
      return store.allEmails.count { $0.analysisState == .pending }
    case .reviewed:
      return store.suggestions.count { $0.status.isReviewed }
    }
  }
}

File: android-native/app/src/main/java/app/dimo/android/features/email/EmailScreen.kt
package app.dimo.android.features.email

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.layout.widthIn
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Cloud
import androidx.compose.material.icons.filled.MailOutline
import androidx.compose.material.icons.filled.Refresh
import androidx.compose.material.icons.filled.Settings
import androidx.compose.material3.AlertDialog
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import app.dimo.android.design.ActionButton
import app.dimo.android.design.ActionButtonVariant
import app.dimo.android.design.DimoColors
import app.dimo.android.design.DimoFont
import app.dimo.android.data.model.EmailAnalysisProvider
import app.dimo.android.data.model.OpenRouterAccessMode

/**
 * Port of `ios-native/Dimo/Features/Email/EmailScreen.swift`.
 *
 * The account strip and filter chips are pinned; only the results scroll, so the
 * user never loses the refresh affordance while paging through suggestions.
 */
@Composable
fun EmailScreen(
  store: EmailFeatureStore,
  onOpenSettings: () -> Unit,
  modifier: Modifier = Modifier,
) {
  Column(
    modifier = modifier
      .fillMaxSize()
      .background(DimoColors.canvas),
  ) {
    EmailHeader(store = store, onOpenSettings = onOpenSettings)

    if (store.accounts.isEmpty()) {
      LazyColumn(
        contentPadding = androidx.compose.foundation.layout.PaddingValues(
          start = 22.dp,
          end = 22.dp,
          top = 10.dp,
          bottom = 34.dp,
        ),
      ) {
        item { ConnectAccountCard(store) }
      }
    } else {
      Column(
        verticalArrangement = Arrangement.spacedBy(16.dp),
        modifier = Modifier
          .fillMaxWidth()
          .padding(horizontal = 22.dp)
          .padding(top = 10.dp, bottom = 12.dp),
      ) {
        AccountAndAnalyzerStrip(store, onOpenSettings)
        EmailFilters(store)
      }

      LazyColumn(
        verticalArrangement = Arrangement.spacedBy(16.dp),
        contentPadding = androidx.compose.foundation.layout.PaddingValues(
          start = 22.dp,
          end = 22.dp,
          bottom = 34.dp,
        ),
      ) {
        if (store.selectedFilter.displaysMessages) {
          if (store.filteredEmails.isEmpty()) {
            item { EmailEmptyState(store.selectedFilter) }
          } else {
            // Suggestion and message rows share the email key as their identity,
            // so the key is namespaced per card type to avoid reuse across tabs.
            items(store.filteredEmails, key = { "message:${it.id}" }) { email ->
              EmailMessageStatusCard(
                email = email,
                onOpen = { store.presentEmail(email.id) },
                onRestore = if (email.analysisState == EmailUIMessageAnalysisState.DISMISSED) {
                  { store.restoreSuggestion(email.id) }
                } else {
                  null
                },
                onRetry = if (email.analysisState == EmailUIMessageAnalysisState.FAILED) {
                  { store.retryAnalysis(email.id) }
                } else {
                  null
                },
              )
            }
          }
        } else if (store.filteredSuggestions.isEmpty()) {
          item { EmailEmptyState(store.selectedFilter) }
        } else {
          items(store.filteredSuggestions, key = { "suggestion:${it.id}" }) { suggestion ->
            EmailSuggestionCard(
              suggestion = suggestion,
              activeCurrency = store.activeCurrency,
              onOpen = { store.presentSources(suggestion) },
              onReview = { store.review(suggestion) },
              onDismiss = { store.dismissSuggestion(suggestion) },
              onRestore = if (suggestion.status == EmailUISuggestionStatus.DISMISSED) {
                { store.restoreSuggestion(suggestion) }
              } else {
                null
              },
              onLinkLate = suggestion.lateMatch?.let { { store.linkLateSuggestion(suggestion) } },
              onKeepLateSeparate = suggestion.lateMatch?.let {
                { store.keepLateSuggestionSeparate(suggestion) }
              },
            )
          }
        }
        item { PrivacyNote(store) }
      }
    }
  }

  store.emailDetail?.let { EmailDetailSheet(it, store::dismissEmailDetail) }
  store.sourceEmailsPresentation?.let { presentation ->
    SourceEmailsSheet(
      presentation = presentation,
      onClose = store::dismissSourceEmails,
      onShowSeparately = if (presentation.canSeparate) {
        {
          val suggestion = presentation.suggestion
          store.dismissSourceEmails()
          store.separateSuggestion(suggestion)
        }
      } else {
        null
      },
    )
  }
  store.purchaseReview?.let { draft ->
    EmailPurchaseReviewSheet(store, draft) { store.purchaseReview = null }
  }
  store.refundReview?.let { review ->
    RefundReviewSheet(
      review = review,
      activeCurrency = store.activeCurrency,
      onCancel = { store.refundReview = null },
      onMarkReviewed = {
        store.dismissSuggestion(it.suggestionId)
        store.refundReview = null
      },
      onConfirm = { store.applyFullRefund(it) },
    )
  }

  store.lastActionError?.let { message ->
    AlertDialog(
      onDismissRequest = store::clearError,
      title = { Text("Email action failed", style = DimoFont.display(17f, FontWeight.SemiBold)) },
      text = { Text(message, style = DimoFont.body(14f)) },
      confirmButton = {
        val accountId = store.pendingReconnectAccountId
        if (accountId != null) {
          TextButton(onClick = {
            store.clearError()
            store.reconnectAccount(accountId)
          }) { Text("Reconnect Gmail") }
        } else {
          TextButton(onClick = store::clearError) { Text("OK") }
        }
      },
      dismissButton = if (store.pendingReconnectAccountId != null) {
        { TextButton(onClick = store::clearError) { Text("OK") } }
      } else {
        null
      },
      containerColor = DimoColors.surface,
    )
  }
}

@Composable
private fun EmailHeader(store: EmailFeatureStore, onOpenSettings: () -> Unit) {
  Row(
    verticalAlignment = Alignment.CenterVertically,
    horizontalArrangement = Arrangement.spacedBy(12.dp),
    modifier = Modifier
      .fillMaxWidth()
      .statusBarsPadding()
      .heightIn(min = 64.dp)
      .padding(horizontal = 22.dp)
      .padding(top = 8.dp),
  ) {
    Column(modifier = Modifier.weight(1f)) {
      Text("Email", style = DimoFont.display(24f, FontWeight.SemiBold), color = DimoColors.ink)
      Text(
        "Purchase and refund suggestions",
        style = DimoFont.body(12f),
        color = DimoColors.muted,
      )
    }
    if (store.hasFailedAnalyses) {
      IconSquareButton(
        icon = Icons.Filled.Refresh,
        contentDescription = if (store.isReanalyzing) {
          "Reanalysing failed emails"
        } else {
          "Reanalyse failed emails"
        },
        enabled = !store.isReanalyzing,
        onClick = store::reanalyzeAllEmails,
      )
    }
    IconSquareButton(
      icon = Icons.Filled.Settings,
      contentDescription = "Email accounts and analyzer settings",
      onClick = onOpenSettings,
    )
  }
}

@Composable
private fun IconSquareButton(
  icon: ImageVector,
  contentDescription: String,
  onClick: () -> Unit,
  enabled: Boolean = true,
) {
  Box(
    contentAlignment = Alignment.Center,
    modifier = Modifier
      .size(42.dp)
      .clip(RoundedCornerShape(13.dp))
      .background(DimoColors.surface)
      .border(1.dp, DimoColors.line, RoundedCornerShape(13.dp))
      .clickable(enabled = enabled, onClick = onClick),
  ) {
    Icon(
      imageVector = icon,
      contentDescription = contentDescription,
      tint = if (enabled) DimoColors.ink else DimoColors.disabled,
      modifier = Modifier.size(18.dp),
    )
  }
}

@Composable
private fun ConnectAccountCard(store: EmailFeatureStore) {
  Column(
    verticalArrangement = Arrangement.spacedBy(14.dp),
    modifier = Modifier
      .fillMaxWidth()
      .clip(RoundedCornerShape(20.dp))
      .background(DimoColors.surface)
      .border(1.dp, DimoColors.line, RoundedCornerShape(20.dp))
      .padding(20.dp),
  ) {
    Box(
      contentAlignment = Alignment.Center,
      modifier = Modifier
        .size(48.dp)
        .clip(RoundedCornerShape(15.dp))
        .background(DimoColors.greenSoft),
    ) {
      Icon(
        Icons.Filled.MailOutline,
        contentDescription = null,
        tint = DimoColors.green,
        modifier = Modifier.size(24.dp),
      )
      Box(
        modifier = Modifier
          .align(Alignment.TopEnd)
          .padding(top = 8.dp, end = 8.dp)
          .size(6.dp)
          .clip(CircleShape)
          .background(DimoColors.green),
      )
    }
    Text(
      "Find expenses in Gmail",
      style = DimoFont.display(20f, FontWeight.SemiBold),
      color = DimoColors.ink,
    )
    Text(
      "Connect one or more Gmail accounts. Dimo reads the latest " +
        "${store.syncWindow.title} directly on this device. Configure OpenRouter in " +
        "settings for analysis; analyzed suggestions sync through Dimo for restore.",
      style = DimoFont.body(14f),
      color = DimoColors.body,
    )
    ActionButton(
      title = "Connect Gmail",
      onClick = store::connectAccount,
      variant = ActionButtonVariant.Accent,
    )
    Text(
      "Read-only Gmail access. Gmail credentials stay on this device. You approve every " +
        "transaction change.",
      style = DimoFont.body(11f),
      color = DimoColors.muted,
      textAlign = TextAlign.Center,
      modifier = Modifier.fillMaxWidth(),
    )
  }
}

@Composable
private fun AccountAndAnalyzerStrip(store: EmailFeatureStore, onOpenSettings: () -> Unit) {
  Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
    LazyRow(
      horizontalArrangement = Arrangement.spacedBy(8.dp),
      modifier = Modifier.weight(1f),
    ) {
      items(store.accounts, key = { it.id }) { account ->
        val needsAttention = account.syncState == EmailUIAccountSyncState.FAILED ||
          account.syncState == EmailUIAccountSyncState.NEEDS_RECONNECT
        Row(
          verticalAlignment = Alignment.CenterVertically,
          horizontalArrangement = Arrangement.spacedBy(8.dp),
          modifier = Modifier
            .height(52.dp)
            .clip(RoundedCornerShape(13.dp))
            .background(DimoColors.surface)
            .border(1.dp, DimoColors.line, RoundedCornerShape(13.dp))
            .clickable(enabled = account.syncState != EmailUIAccountSyncState.SYNCING) {
              if (account.syncState == EmailUIAccountSyncState.NEEDS_RECONNECT) {
                store.reconnectAccount(account.id)
              } else {
                store.refreshAccount(account.id)
              }
            }
            .padding(horizontal = 10.dp),
        ) {
          Box(
            contentAlignment = Alignment.Center,
            modifier = Modifier
              .size(28.dp)
              .clip(RoundedCornerShape(9.dp))
              .background(
                if (needsAttention) DimoColors.dangerSoft else DimoColors.greenSoft,
              ),
          ) {
            if (account.syncState == EmailUIAccountSyncState.SYNCING) {
              CircularProgressIndicator(
                strokeWidth = 2.dp,
                color = DimoColors.green,
                modifier = Modifier.size(14.dp),
              )
            } else {
              Icon(
                Icons.Filled.Refresh,
                contentDescription = null,
                tint = if (needsAttention) DimoColors.danger else DimoColors.green,
                modifier = Modifier.size(14.dp),
              )
            }
          }
          Column(modifier = Modifier.widthIn(max = 168.dp)) {
            Text(
              account.emailAddress,
              style = DimoFont.body(11f, FontWeight.Medium),
              color = DimoColors.ink,
              maxLines = 1,
              overflow = TextOverflow.Ellipsis,
            )
            Text(
              account.statusDetail ?: account.syncState.title,
              style = DimoFont.body(10f),
              color = if (needsAttention) DimoColors.danger else DimoColors.muted,
              maxLines = 1,
              overflow = TextOverflow.Ellipsis,
            )
          }
        }
      }
    }

    Row(
      verticalAlignment = Alignment.CenterVertically,
      horizontalArrangement = Arrangement.spacedBy(8.dp),
      modifier = Modifier
        .weight(1f)
        .height(52.dp)
        .clip(RoundedCornerShape(13.dp))
        .background(DimoColors.surface)
        .border(1.dp, DimoColors.line, RoundedCornerShape(13.dp))
        .clickable(onClick = onOpenSettings)
        .padding(horizontal = 10.dp),
    ) {
      Box(
        contentAlignment = Alignment.Center,
        modifier = Modifier
          .size(28.dp)
          .clip(RoundedCornerShape(9.dp))
          .background(
            if (store.selectedProvider != null) DimoColors.greenSoft else DimoColors.canvasDeep,
          ),
      ) {
        Icon(
          Icons.Filled.Cloud,
          contentDescription = null,
          tint = if (store.selectedProvider != null) DimoColors.green else DimoColors.muted,
          modifier = Modifier.size(14.dp),
        )
      }
      Column(modifier = Modifier.weight(1f)) {
        Text(
          store.activeAnalyzerTitle,
          style = DimoFont.body(11f, FontWeight.SemiBold),
          color = DimoColors.ink,
          maxLines = 1,
          overflow = TextOverflow.Ellipsis,
        )
        Text(
          store.analysisStatusDetail,
          style = DimoFont.body(10f),
          color = DimoColors.muted,
          maxLines = 1,
          overflow = TextOverflow.Ellipsis,
        )
      }
    }
  }
}

@Composable
private fun EmailFilters(store: EmailFeatureStore) {
  LazyRow(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
    items(EmailSuggestionFilter.entries.toList(), key = { it.name }) { filter ->
      val selected = store.selectedFilter == filter
      val count = filterCount(store, filter)
      Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        modifier = Modifier
          .height(38.dp)
          .clip(RoundedCornerShape(50))
          .background(if (selected) DimoColors.ink else DimoColors.surface)
          .then(
            if (selected) {
              Modifier
            } else {
              Modifier.border(1.dp, DimoColors.line, RoundedCornerShape(50))
            },
          )
          .clickable { store.setFilter(filter) }
          .padding(horizontal = 14.dp),
      ) {
        Text(
          filter.title,
          style = DimoFont.body(12f, FontWeight.Medium),
          color = if (selected) DimoColors.canvas else DimoColors.ink,
        )
        if (count > 0) {
          Text(
            count.toString(),
            style = DimoFont.body(10f, FontWeight.Bold),
            color = if (selected) DimoColors.canvas else DimoColors.ink,
            modifier = Modifier
              .clip(RoundedCornerShape(50))
              .background(
                if (selected) {
                  DimoColors.canvas.copy(alpha = 0.2f)
                } else {
                  DimoColors.canvasDeep
                },
              )
              .padding(horizontal = 5.dp, vertical = 2.dp),
          )
        }
      }
    }
  }
}

private fun filterCount(store: EmailFeatureStore, filter: EmailSuggestionFilter): Int =
  when (filter) {
    EmailSuggestionFilter.ALL -> store.allEmails.size
    EmailSuggestionFilter.PURCHASES -> store.pendingPurchaseCount
    EmailSuggestionFilter.REFUNDS -> store.suggestions.count {
      it.status == EmailUISuggestionStatus.PENDING_REFUND &&
        it.kind == EmailUISuggestionKind.REFUND
    }

    EmailSuggestionFilter.AWAITING_ANALYSIS -> store.allEmails.count {
      it.analysisState == EmailUIMessageAnalysisState.PENDING
    }

    EmailSuggestionFilter.ERRORS -> store.analysisErrorCount
    EmailSuggestionFilter.REVIEWED -> store.suggestions.count { it.status.isReviewed }
  }

@Composable
private fun EmailEmptyState(filter: EmailSuggestionFilter) {
  val (title, detail) = when (filter) {
    EmailSuggestionFilter.ALL -> "No scanned emails yet" to
      "Scanned messages and their local analysis status will appear here."

    EmailSuggestionFilter.PURCHASES -> "No purchase suggestions" to
      "Email scanning is best effort and is not real time. Tap an account above to refresh."

    EmailSuggestionFilter.REFUNDS -> "No refund suggestions" to
      "Email scanning is best effort and is not real time. Tap an account above to refresh."

    EmailSuggestionFilter.AWAITING_ANALYSIS -> "No emails awaiting analysis" to
      "Fetched emails waiting for analysis will appear here."

    EmailSuggestionFilter.ERRORS -> "No analysis errors" to
      "Emails that could not be analyzed will appear here with retry options."

    EmailSuggestionFilter.REVIEWED -> "Nothing reviewed yet" to
      "Suggestions you add, dismiss, or apply will appear here."
  }
  Column(
    horizontalAlignment = Alignment.CenterHorizontally,
    verticalArrangement = Arrangement.spacedBy(10.dp),
    modifier = Modifier
      .fillMaxWidth()
      .padding(horizontal = 24.dp, vertical = 44.dp),
  ) {
    Text(title, style = DimoFont.body(15f, FontWeight.SemiBold), color = DimoColors.ink)
    Text(
      detail,
      style = DimoFont.body(13f),
      color = DimoColors.muted,
      textAlign = TextAlign.Center,
    )
  }
}

@Composable
private fun PrivacyNote(store: EmailFeatureStore) {
  val text = when (store.selectedProvider) {
    EmailAnalysisProvider.OPEN_ROUTER -> when (store.openRouterAccessMode) {
      OpenRouterAccessMode.FREE_SHARED ->
        "Free-model analysis goes from this device through Dimo's servers to OpenRouter. " +
          "Analyzed suggestions and their email text then sync through Dimo so they restore " +
          "across your devices. You still approve every transaction change."

      OpenRouterAccessMode.BRING_YOUR_OWN_KEY ->
        "Analysis goes from this device to OpenRouter and the selected provider. Analyzed " +
          "suggestions and their email text then sync through Dimo so they restore across " +
          "your devices. You still approve every transaction change."
    }

    null ->
      "Gmail credentials and pending email content stay on this device until you configure " +
        "OpenRouter. Analyzed suggestions later sync through Dimo for restore."
  }
  Text(
    text = text,
    style = DimoFont.body(11f),
    color = DimoColors.muted,
    modifier = Modifier.padding(top = 4.dp),
  )
}
